docs/src/model_creation/functional_parameters.md

```diff
@@ -0,0 +1,148 @@
+```@example functional_parameters_circ_rhythm
+input = light_in(t)
+rs_dsl = @reaction_network rs begin
+    (kA*$input, kD), Pᵢ <--> Pₐ
+end
+```
+We can confirm that this model is identical to our programmatic one (and should we wish to, we can simulate it using identical syntax syntax).
+```@example functional_parameters_circ_rhythm
+rs == rs_dsl
+```
+
+## [Non-time functional parameters](@id functional_parameters_sir)
+Previously we have demonstrated functional parameters that are a function of time. However, functional parameters can be functions of any variable (however, currently, more than one argument is not supported). Here we will demonstrate this using a [SIR model](@ref basic_CRN_library_sir), but instead of having the infection rate scale linearly with the number of infected individuals, we instead assume we have measured data of the infection rate (as dependent on the number of infected individuals) and wish to use this instead. Normally we use the following infection reaction in the SIR model:
+```julia
+@reaction k1, S + I --> 2I
+```
+In practise, this is identical to
+```julia
+@reaction k1*I, S --> I
+```
+Due to performance reasons (especially for jump simulations) the former approach is *strongly* encouraged. Here, however, we will assume that we have measured real data of how the number of infected individuals affects the infection rate, and wish to use this in our model, i.e. something like this:
+```julia
+@reaction k1*i_rate(I), S --> I
+```
+This is a case where we can use a functional parameter, whose value depends on the value of $I$.
+
+We start by declaring the functional parameter that describes how the infection rate depends on the number of infected individuals. We also plot the measured infection rate, and compare it to the theoretical rate usually used in the SIR model.
+```@example functional_parameters_sir
+using DataInterpolations, Plots
+I_grid = collect(0.0:5.0:100.0)
+I_meassured = 300.0 *(0.8*rand(length(I_grid)) .+ 0.6) .* I_grid ./ (300 .+ I_grid)
+I_rate = LinearInterpolation(I_meassured, I_grid)
+plot(I_rate; label = "Meassured infection rate")
+plot!(I_grid, I_grid; label = "Normal SIR infection rate")
+```
+Next, we create our model (using the DSL-approach). As `I_rate` will be a function of $I$ we will need to declare this species first as well.
+```@example functional_parameters_sir
+using Catalyst
+@species I(default_t())
+inf_rate = I_rate(I)
+sir = @reaction_network rs begin
+    k1*$inf_rate, S --> I
+    k2, I --> R
+end
+nothing # hide
+```
+Finally, we can simualte our model.
+```@example functional_parameters_sir
+using OrdinaryDiffEqDefault
+u0 = [:S => 99.0, :I => 1.0, :R => 0.0]
+ps = [:k1 => 0.002, :k2 => 0.01, inf_rate => I_rate]
```

Comment:
Again, why not
```suggestion
ps = [:k1 => 0.002, :k2 => 0.01, :inf_rate => I_rate]
```

Reply:
This is not directly possible in this case, but I can rewrite a bit to enable it.

Reply:
fixed